src/test/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtilTest.java

```diff
@@ -28,29 +29,77 @@
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+import org.itsallcode.io.Capturable;
+import org.itsallcode.junit.sysextensions.SystemOutGuard;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 
+import com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport;
 import com.puppycrawl.tools.checkstyle.internal.utils.CheckUtil;
 
-public final class MetadataGeneratorUtilTest {
+@ExtendWith(SystemOutGuard.class)
+public final class MetadataGeneratorUtilTest extends AbstractModuleTestSupport {
 
     private final List<String> modulesContainingNoMetadataFile = Arrays.asList(
             "Checker",
             "TreeWalker",
             "JavadocMetadataScraper"
     );
 
+    @Override
+    protected String getPackageLocation() {
+        return null;
+    }
+
+    /**
+     * Generates metadata for checkstyle modules and verifies number of
+     * generated metadata modules match the number of checkstyle modules.
+     * Also verifies whether every checkstyle module contains description.
+     *
+     * @param systemOut wrapper for {@code System.out}
+     * @throws Exception if exception occurs during generating metadata or
+     *                   if an I/O error is thrown when accessing the starting file.
+     * @noinspection UseOfSystemOutOrSystemErr
+     */
     @Test
-    public void testMetadataFilesGenerationAllFiles() throws Exception {
+    public void testMetadataFilesGenerationAllFiles(@SystemOutGuard.SysOut Capturable systemOut)
+            throws Exception {
+        systemOut.captureMuted();
 
         MetadataGeneratorUtil.generate(System.getProperty("user.dir")
                         + "/src/main/java/com/puppycrawl/tools/checkstyle",
-                "checks", "filters", "filefilters");
-        final Set<String> metaFiles;
+                System.out, "checks", "filters", "filefilters");
 
+        final String[] expectedErrorMessages = {
+            "31: " + getCheckMessage(MSG_DESC_MISSING, "AbstractSuperCheck"),
+            "45: " + getCheckMessage(MSG_DESC_MISSING, "AbstractHeaderCheck"),
+            "42: " + getCheckMessage(MSG_DESC_MISSING, "AbstractJavadocCheck"),
+            "45: " + getCheckMessage(MSG_DESC_MISSING, "AbstractClassCouplingCheck"),
+            "26: " + getCheckMessage(MSG_DESC_MISSING, "AbstractAccessControlNameCheck"),
+            "30: " + getCheckMessage(MSG_DESC_MISSING, "AbstractNameCheck"),
+            "29: " + getCheckMessage(MSG_DESC_MISSING, "AbstractParenPadCheck"),
+        };
+
+        final String[] actualViolations = systemOut.getCapturedData().split("\\n");
+        final Pattern pattern = Pattern.compile("((?<=:)\\d.*:.*(?=\\s\\[))");
```

Comment:
Can you come up with some more descriptive name for this pattern, or add a comment here?

Reply:
changed to `violationExtractingPattern` fine?

Reply:
Resolved.